hrjobcontract/tests/phpunit/CRM/Hrjobcontract/BAO/HRJobDetailsTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+
+/**
+ * Class CRM_Hrjobcontract_BAO_HRJobDetailsTest
+ *
+ * @group headless
+ */
+class CRM_Hrjobcontract_BAO_HRJobDetailsTest extends PHPUnit_Framework_TestCase implements
+  HeadlessInterface,
+  TransactionalInterface {
+
+  use HRJobContractTestTrait;
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()->installMe(__DIR__)->apply();
+  }
+
+  function setUp() {
+    $upgrader = CRM_Hrjobcontract_Upgrader::instance();
+    $upgrader->install();
+  }
+
+  function testValidateDatesWithDifferentDateSets() {
+    // Create test Contact.
+    $contactParams = array("first_name" => "chrollo", "last_name" => "lucilfer");
+    $contactID =  $this->createContact($contactParams);
+
+    // Create Two test Job Contracts.
+    $this->createJobContract($contactID, '2016-01-01', '2016-01-10');
+    $this->createJobContract($contactID, '2016-02-01');
+
+    // Set of tests against invalid dates period (there are Job Contracts
+    // already with period dates overlapping.
+
+    $sampleFailedDates = [
+      ['period_start_date' => '2016-01-01', 'period_end_date'=> '2016-10-10'],
+      ['period_start_date' => '2016-01-03', 'period_end_date'=> '2016-01-05'],
+      ['period_start_date' => '2016-01-01', 'period_end_date'=> '2016-01-02'],
+      ['period_start_date' => '2016-01-31', 'period_end_date'=> '2016-02-01'],
+      ['period_start_date' => '2016-02-10', 'period_end_date'=> '2016-02-20'],
+      ['period_start_date' => '2015-01-01'],
+      ['period_start_date' => '2016-03-01'],
+    ];
+
+    foreach ($sampleFailedDates as $datePair)  {
+      $params = ['contact_id' => $contactID, 'period_start_date' => $datePair['period_start_date']];
+      if (!empty($datePair['period_end_date']))  {
+        $params['period_end_date'] = $datePair['period_end_date'];
+      }
+
+      $result = CRM_Hrjobcontract_BAO_HRJobDetails::validateDates($params);
+
+      $this->assertFalse($result['success']);
+    }
+
+    // Now we test against valid dates so we expect the result of
+    // 'validateDates' call to be TRUE.
```

Comment:
well I wasn't really certain what do in this case , since I initialy thought to create a seperate test case for each start,end date pair  but that would be too much and choosing names for test cases will be a bit tricky .. but separating them into two test methods sounds like a good option.